src/compas/geometry/xforms/quaternion.py

```diff
@@ -0,0 +1,118 @@
+from compas.geometry.transformations import quaternion_multiply
+from compas.geometry.transformations import quaternion_conjugate
+from compas.geometry.transformations import quaternion_unitize
+from compas.geometry.transformations import quaternion_canonic
+from compas.geometry.transformations import quaternion_norm
+from compas.geometry.transformations import quaternion_is_unit
+
+__all__ = ['Quaternion']
+
+class Quaternion(object):
+    def __init__(self, w,x,y,z):
+
+        # TODO: add checks for value type
+
+        self.w = w
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def __iter__(self):
+        return iter([self.w, self.x, self.y, self.z])
+
+    def __str__(self):
+        return "Quaternion = %s" % list(self)
+
+    def __mul__(self, other):
+        """Multiply operator for two quaternions.
+
+        Parameters
+        ----------
+        R
+            First argument, a Quaternion object.
+        Q
+            Second argument, a Quaternion object.
+
+        Returns
+        -------
+        Quaternion
+            P = R * Q
+
+        Examples
+        --------
+        >>> R = Quaternion(1,-2,3,-4).unitized
+        >>> Q = Quaternion(0,1,-2,-3).unitized
+        >>> P = R * Q
+        >>> print(P)
+        Quaternion = [-0.19518001458970669, -0.7807200583588265, -0.5855400437691198, -0.09759000729485334]
```

Comment:
Similar here, rather use `allclose`

Reply:
also, the `__rep__` should return something with reasonable precision, for example 4 digits...